3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -265,12 +246,79 @@ object EnrichmentManager {
       }
     }
 
-    // Finally anonymize the IP address
+    // To anonymize the IP address
     Option(event.user_ipaddress).map(ip => event.user_ipaddress = registry.getAnonIpEnrichment match {
       case Some(anon) => anon.anonymizeIp(ip)
       case None => ip
     })
 
+    // Parse the useragent using user-agent-utils
+    val client = {
+      registry.getUserAgentUtilsEnrichment match {
+        case Some(uap) => {
+          Option(event.useragent) match {
+            case Some(ua) =>
+              val ca = uap.extractClientAttributes(ua)
+              ca.flatMap(c => {
+                event.br_name = c.browserName
+                event.br_family = c.browserFamily
+                c.browserVersion.map(bv => event.br_version = bv)
+                event.br_type = c.browserType
+                event.br_renderengine = c.browserRenderEngine
+                event.os_name = c.osName
+                event.os_family = c.osFamily
+                event.os_manufacturer = c.osManufacturer
+                event.dvce_type = c.deviceType
+                event.dvce_ismobile = CU.booleanToJByte(c.deviceIsMobile)
+                c.success
+                })
+              ca
+            case None => unitSuccess // No fields updated
+          }
+        }
+        case None => unitSuccess
+      }
+    }
+
+    // Create the ua_parser_context
+    val uaParser = {
+      registry.getUaParserEnrichment match {
+        case Some(uap) => {
+          Option(event.useragent) match {
+            case Some(ua) => uap.extractUserAgent(ua).map(_.some)
+            case None => None.success // No fields updated
+          }
+        }
+        case None => None.success
+      }
+    }
+
+    // Finalize the currency conversion
+    val currency = {
+      registry.getCurrencyConversionEnrichment match {
+        case Some(currency) => {
+          event.base_currency = currency.baseCurrency
```

Comment:
Could you add a note here something like:

// Note stringToMaybeDouble is applied to either-valid-or-null event POJO properties, so we don't expect a Failure